thrimbletrimmer/scripts/common.js

```diff
@@ -313,6 +313,27 @@ function setUpVideoControls() {
 			globalSeekTimer = null;
 		}
 	});
+
+	updateFrameLink(videoElement.currentTime);
+	videoElement.addEventListener("timeupdate", (_event) => {
+		updateFrameLink(videoElement.currentTime);
+	});
+}
+
+function updateFrameLink(videoTime) {
+	const frameLink = document.getElementById("download-frame");
+	if (frameLink !== undefined) {
+		const dateTime = dateTimeFromVideoPlayerTime(videoTime);
+		if (!dateTime) {
+			frameLink.href = "#";
+		} else {
+			const wubloaderTime = wubloaderTimeFromDateTime(dateTime)
```

Comment:
A few things about this:
- It feels like doing a time conversion on every time update (which, granted, isn't every frame) might be approaching a lot. I don't want to convince the browser to give us fewer updates (assuming this has any influence).
- Following the above, note that the time might be inaccurate due to not getting a timeupdate for every frame. This could be up to a quarter second off (relevant for jump cuts, etc.).

Given that, it might be better if we can do this as a function that runs when the link is clicked to start the download of the relevant image (getting the current timestamp at that point, which we can query). [This thing I just found](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download) might be able to help a lot with that.

Reply:
That seems reasonable, and makes more sense than this approach. I'll see if I can get it done while travelling.